npm/pkg/dataplane/ipsets/ipsetmanager_windows_test.go

```diff
@@ -0,0 +1,370 @@
+	iMgr := NewIPSetManager(iMgrApplyAlwaysCfg, io)
+
+	iMgr.CreateIPSets([]*IPSetMetadata{TestNSSet.Metadata})
+	require.NoError(t, iMgr.AddToSets([]*IPSetMetadata{TestNSSet.Metadata}, "10.0.0.0", "a"))
+	require.NoError(t, iMgr.AddToSets([]*IPSetMetadata{TestNSSet.Metadata}, "10.0.0.1", "b"))
+	iMgr.CreateIPSets([]*IPSetMetadata{TestKeyPodSet.Metadata})
+	require.NoError(t, iMgr.AddToSets([]*IPSetMetadata{TestKeyPodSet.Metadata}, "10.0.0.5", "c"))
+	iMgr.CreateIPSets([]*IPSetMetadata{TestKVPodSet.Metadata})
+	iMgr.CreateIPSets([]*IPSetMetadata{TestNamedportSet.Metadata})
+	iMgr.CreateIPSets([]*IPSetMetadata{TestCIDRSet.Metadata})
+	iMgr.CreateIPSets([]*IPSetMetadata{TestKeyNSList.Metadata})
+	require.NoError(t, iMgr.AddToLists([]*IPSetMetadata{TestKeyNSList.Metadata}, []*IPSetMetadata{TestNSSet.Metadata, TestKeyPodSet.Metadata}))
+	iMgr.CreateIPSets([]*IPSetMetadata{TestKVNSList.Metadata})
+	require.NoError(t, iMgr.AddToLists([]*IPSetMetadata{TestKVNSList.Metadata}, []*IPSetMetadata{TestKVPodSet.Metadata}))
+	iMgr.CreateIPSets([]*IPSetMetadata{TestNestedLabelList.Metadata})
+	toAddOrUpdateSetMap := map[string]hcn.SetPolicySetting{
+		TestNSSet.PrefixName: {
+			Id:         TestNSSet.HashedName,
+			PolicyType: hcn.SetPolicyTypeIpSet,
+			Name:       TestNSSet.PrefixName,
+			Values:     "10.0.0.0,10.0.0.1",
+		},
+		TestKeyPodSet.PrefixName: {
+			Id:         TestKeyPodSet.HashedName,
+			PolicyType: hcn.SetPolicyTypeIpSet,
+			Name:       TestKeyPodSet.PrefixName,
+			Values:     "10.0.0.5",
+		},
+		TestKVPodSet.PrefixName: {
+			Id:         TestKVPodSet.HashedName,
+			PolicyType: hcn.SetPolicyTypeIpSet,
+			Name:       TestKVPodSet.PrefixName,
+			Values:     "",
+		},
+		TestNamedportSet.PrefixName: {
+			Id:         TestNamedportSet.HashedName,
+			PolicyType: hcn.SetPolicyTypeIpSet,
+			Name:       TestNamedportSet.PrefixName,
+			Values:     "",
+		},
+		TestCIDRSet.PrefixName: {
+			Id:         TestCIDRSet.HashedName,
+			PolicyType: hcn.SetPolicyTypeIpSet,
+			Name:       TestCIDRSet.PrefixName,
+			Values:     "",
+		},
+		TestKeyNSList.PrefixName: {
+			Id:         TestKeyNSList.HashedName,
+			PolicyType: SetPolicyTypeNestedIPSet,
+			Name:       TestKeyNSList.PrefixName,
+			Values:     fmt.Sprintf("%s,%s", TestNSSet.HashedName, TestKeyPodSet.HashedName),
+		},
+		TestKVNSList.PrefixName: {
+			Id:         TestKVNSList.HashedName,
+			PolicyType: SetPolicyTypeNestedIPSet,
+			Name:       TestKVNSList.PrefixName,
+			Values:     TestKVPodSet.HashedName,
+		},
+		TestNestedLabelList.PrefixName: {
+			Id:         TestNestedLabelList.HashedName,
+			PolicyType: SetPolicyTypeNestedIPSet,
+			Name:       TestNestedLabelList.PrefixName,
+			Values:     "",
+		},
+	}
+	toAddOrUpdateSetNames := make([]string, 0, len(toAddOrUpdateSetMap))
+	for setName := range toAddOrUpdateSetMap {
+		toAddOrUpdateSetNames = append(toAddOrUpdateSetNames, setName)
+	}
+
+	assertEqualContentsTestHelper(t, toAddOrUpdateSetNames, iMgr.toAddOrUpdateCache)
```

Comment:
mentioned offline, but we should just use the cache in OS-specific UTs and assert correct cache contents in generic ipsetmanager